src/main/java/com/microsoft/spring/data/gremlin/exception/IllegalGremlinConfigurationException.java

```diff
@@ -0,0 +1,13 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.exception;
+
+public class IllegalGremlinConfigurationException extends IllegalArgumentException {
+
+    public IllegalGremlinConfigurationException(String msg) {
+        super(msg);
+    }
```

Comment:
A new exception constructor might be added. IllegalGremlinConfigurationException(msg, exception)